### PR TITLE
CPLP-1660 subscription log

### PR DESCRIPTION
**Pull request by Phil91**
Added logging to autoSetup on subscription
Added Endpoint for Apps AutoSetup

### CODE REVIEW

File: src/marketplace/Services.Service/Controllers/ServicesController.cs [hunk 1]
```diff
@@ -93,7 +93,7 @@ public async Task<CreatedAtRouteResult> CreateServiceOffering([FromBody] Offerin
     /// <response code="404">No Service was found for the given id.</response>
     [HttpPost]
     [Route("{serviceId}/subscribe")]
-    [Authorize(Roles = "subscribe_service")]
+    //[Authorize(Roles = "subscribe_service")]
```
Comment (ntruchsess) [inline src/marketplace/Services.Service/Controllers/ServicesController.cs:96]: I guess this has been removed for testing only? Please re-enable role-based authorisation-check

**Review by ntruchsess — CHANGES_REQUESTED** [review]
as I thought I would merge this right away I did some cosmetical changes. Didn't want to re-enable the role-check though as I was not sure whether this might have been explicitly requested by Julia for testing. Please pull before proceeding.

File: src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs [hunk 2]
```diff
@@ -77,6 +79,7 @@ public async Task<Guid> AddServiceSubscription(Guid serviceId, string iamUserId,
             }
             catch (Exception e)
             {
+                _logger.LogInformation(e.Message);
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs]: i provided a logging template to have a nicer information

File: src/marketplace/Services.Service/Controllers/ServicesController.cs (hunk 1 above)
Comment (Phil91) [reply src/marketplace/Services.Service/Controllers/ServicesController.cs]: jep, sorry for missing - re-enabled

File: tests/marketplace/Apps.Service.Tests/BusinessLogic/AppBusinessLogicTests.cs [hunk 3]
```diff
@@ -133,6 +137,27 @@ public async Task AddCompanyAppSubscription_ExecutesSuccessfully()
...
+    {
+        // Arrange
+        var offerService = A.Fake<IOfferService>();
+        A.CallTo(() => offerService.AutoSetupServiceAsync(A<OfferAutoSetupData>._, A<IDictionary<string, IEnumerable<string>>>._, A<IDictionary<string, IEnumerable<string>>>._, A<string>._, A<OfferTypeId>._))
+            .ReturnsLazily(() => new OfferAutoSetupResponseData(new TechnicalUserInfoData(Guid.NewGuid(), "abcSecret", "sa1"), new ClientInfoData(Guid.NewGuid().ToString())));
+        var data = new OfferAutoSetupData(Guid.NewGuid(), "https://www.offer.com");
+        var sut = _fixture.Create<AppsBusinessLogic>();
+
+        // Act
+        var result = await sut.AutoSetupAppAsync(data, IamUserId).ConfigureAwait(false);
```
Comment (ntruchsess) [inline tests/marketplace/Apps.Service.Tests/BusinessLogic/AppBusinessLogicTests.cs:153]: as the offerservice is not injected this call is not testing anything. If the result would have been tested for correctness, you would have noticed:
```
    public async Task AutoSetupService_ReturnsExcepted()
    {
        var offerAutoSetupResponseData = _fixture.Create<OfferAutoSetupResponseData>();
        // Arrange
        var offerService = A.Fake<IOfferService>();
        A.CallTo(() => offerService.AutoSetupServiceAsync(A<OfferAutoSetupData>._, A<IDictionary<string, IEnumerable<string>>>._, A<IDictionary<string, IEnumerable<string>>>._, A<string>._, A<OfferTypeId>._))
            .ReturnsLazily(() => offerAutoSetupResponseData);
        var data = new OfferAutoSetupData(Guid.NewGuid(), "https://www.offer.com");
//        _fixture.Inject(offerService); // when doing the service-injection the test passes.

        var sut = _fixture.Create<AppsBusinessLogic>();

        // Act
        var result = await sut.AutoSetupAppAsync(data, IamUserId).ConfigureAwait(false);

        // Assert
        result.Should().Be(offerAutoSetupResponseData);
    }
```

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**